src/v/cluster/tx_gateway_frontend.h

```diff
@@ -25,6 +25,9 @@
 
 namespace cluster {
 
+using clock_type = ss::lowres_clock;
+using time_point_type = clock_type::time_point;
```

Comment:
We should avoid defining generic types that are used in a context specific way (like `clock_type` here) at this high-level scope in a header. I'd recommend moving them down into the tx_gateway_frontend scope.